### PR TITLE
feat(private-endpoint): add Private endpoint configuration

### PR DESCRIPTION
**Pull request by qprichard**

### CODE REVIEW

File: network.tf [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+  name                 = var.name
+  resource_group_name  = var.resource_group.name
+  virtual_network_name = var.subnet.virtual_network.name
+
+  enforce_private_link_endpoint_network_policies = true
+
+  address_prefixes = var.subnet.address_prefixes
+}
+
```
Comment (Laudenlaruto) [inline network.tf:12]: It doesn't seem right to create the module in the module. You can share a subnet with private endpoint of multiple Azure resources

File: variables.tf [hunk 2]
```diff
@@ -93,14 +100,37 @@ variable "access_policy" {
...
+variable "subnet" {
+  description = "The subnet configuration for private endpoint."
+  type = object({
+    address_prefixes = list(string)
+    virtual_network = object({
+      name = string
+    })
+  })
+  default = null
+}
```
Comment (Laudenlaruto) [inline variables.tf:124]: Out of scope

File: variables.tf [hunk 3]
```diff
@@ -93,14 +100,37 @@ variable "access_policy" {
   default     = {}
 }
 
+variable "private_endpoint" {
```
Comment (Laudenlaruto) [inline variables.tf:103]: Why not add subnet id to this variable ?

File: network.tf [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+resource "azurerm_private_endpoint" "this" {
+  count = var.private_endpoint.enable ? 1 : 0
+
+  name                = var.name
+  resource_group_name = var.resource_group.name
+  location            = var.resource_group.location
+  subnet_id           = var.subnet != null ? azurerm_subnet.this.0.id : var.subnet_id
+
+  dynamic "private_dns_zone_group" {
+    for_each = length(var.private_endpoint.private_dns_zone_ids) > 0 ? [1] : [0]
```
Comment (Laudenlaruto) [inline network.tf:22]: Not useful if the variable is not optional

File: variables.tf [hunk 5]
```diff
@@ -93,14 +100,37 @@ variable "access_policy" {
...
+  description = "The private endpoint configuration."
+  type = object({
+    enable               = bool,
+    private_dns_zone_ids = list(string)
```
Comment (Laudenlaruto) [inline variables.tf:107]: Why a list if you use only 1 zone ?

**Review by piaverous — APPROVED** [review]

**Pull request merged**